### PR TITLE
[Add][stream] toast

### PR DESCRIPTION
**Pull request by siva2204**
-  add toast for stream updates form server

### CODE REVIEW

File: lib/utils/streams_to_toast.dart [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      var price = update.transaction.price;
+      var total = update.transaction.total;
+      var stockOrStocks = 'stock' + (stockQty.abs() > 1 ? 's' : '');
+
+      switch (update.transaction.type) {
+        case TransactionType.FROM_EXCHANGE_TRANSACTION:
+          showSnackBar(context,
+              'You have bought $stockQty $stockOrStocks of ${stock.fullName} at ₹$price from Exchange',
+              type: SnackBarType.success);
+          break;
```
[the remark this answers is not in the document]
Comment (siva2204) [reply lib/utils/streams_to_toast.dart:130]: They won't clash, but ig its better to have one snackbar, we can handle all the toast here, can we remove it From there?

It'll be easy for us to maintain all toast here

[the remark this answers is not in the document]
Comment (dhruv1294) [reply lib/utils/streams_to_toast.dart:130]: ok

[the remark this answers is not in the document]
Comment (siva2204) [reply lib/utils/streams_to_toast.dart:130]: there wont be any lag, this is like a ws connection, we'll get the updates instantaneously

**Comment by siva2204** [comment]
i think we can go ahead and merge this, if it feels like too many toast, we can reduce them in the near feature

File: lib/utils/streams_to_toast.dart [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+          showSnackBar(context,
+              'You have received ₹$total as dividends from ${stock.fullName}',
+              type: SnackBarType.success);
+          break;
+        default:
+      }
+    }
+  }
+
+  Future.wait([
```
Comment (Sudhindra3000) [inline lib/utils/streams_to_toast.dart:171]: Why do we need Future.wait here. It'll just  keep listening to streams and showing the snackbars right

File: lib/utils/streams_to_toast.dart [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+/// - transaction update
+/// - notification update
+/// - stock bankrupt state
+/// - user blocked state
+/// - daily challenges state
+/// - dividends update
+/// - user(current loggedIn user) blocked state
+void streamToast(BuildContext context) async {
+  // gamestate stream listener
+  Future<void> listenGameStateStream() async {
```
Comment (Sudhindra3000) [inline lib/utils/streams_to_toast.dart:26]: Define these as separate functions, outside of  `streamToast`

Comment (Sudhindra3000) [reply lib/utils/streams_to_toast.dart:26]: And make them private to this file

File: lib/utils/streams_to_toast.dart [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+/// - referral and daily challenge reward claim status
+/// - orders update
+/// - transaction update
+/// - notification update
+/// - stock bankrupt state
+/// - user blocked state
+/// - daily challenges state
+/// - dividends update
+/// - user(current loggedIn user) blocked state
+void streamToast(BuildContext context) async {
```
Comment (Sudhindra3000) [inline lib/utils/streams_to_toast.dart:24]: Can we give a better name? But i can't think of anything

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/utils/streams_to_toast.dart (hunk 4 above)
Comment (siva2204) [reply lib/utils/streams_to_toast.dart:24]: how about this `streamSnackBarUpdates` ?

File: lib/utils/streams_to_toast.dart [hunk 5]
```diff
@@ -21,156 +21,153 @@ import 'package:flutter/widgets.dart';
...
-          }
-          break;
-
-        case GameStateUpdateType.StockDividendStateUpdate:
-          var stock = getIt<GlobalStreams>()
-              .stockMapStream
-              .value[update.gameState.stockDividendState.stockId]!;
-          showSnackBar(context,
-              '${stock.fullName} has earned huge profits and thus has planned to give out dividends to its shareholders soon!');
+void streamSnackBarUpdates(BuildContext context) {
```
Comment (Sudhindra3000) [inline lib/utils/streams_to_toast.dart:24]: Change the file name also

**Review by Sudhindra3000 — APPROVED** [review]

**Pull request merged**